sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor.cs

```diff
@@ -1286,6 +1286,25 @@ protected virtual LastEnqueuedEventProperties ReadLastEnqueuedEventProperties(st
                         Logger.ProcessorStoppingCancellationWarning(Identifier, EventHubName, ConsumerGroup, cancelEx.Message);
                     }
                 }
+
+                // Ensure the load balancing and partition ownership intervals are not configured too close.  The ownership
+                // interval should be at least twice the load balancing interval.  Documented guidance recommends a factor of
+                // three.
+                //
+                // A smaller gap is valid and may be desirable in some unusual scenarios, but is likely to cause
+                // issues for mainline scenarios.  Emit a warning to the error handler and logs, but allow starting to proceed.
+
+                var ownershipSeconds = Options.PartitionOwnershipExpirationInterval.TotalSeconds;
+                var loadBalancingSeconds = Options.LoadBalancingUpdateInterval.TotalSeconds;
+
+                if (ownershipSeconds < (loadBalancingSeconds * 2))
+                {
+                    Logger.ProcessorLoadBalancingIntervalsTooCloseWarning(Identifier, EventHubName, loadBalancingSeconds, ownershipSeconds);
```

Comment:
This seems sort of aggressive if we acknowledge that this could be valid configuration for some scenarios. Is it not sufficient to point to troubleshooting docs when the customer ultimately encounters an exception due to this setup?

Reply:
That's the reason we only warn at startup rather than failing validation or alerting on every load balancing tick.  

The valid scenario would be a very, very rare case with extreme sensitivity around maintaining throughput and supreme confidence that you've got an over-provisioned machine and network such that you never bog down.    The 95% case, this is a problem and one that we've received feedback for "the processor should tell me."